client/core/types.go

```diff
@@ -680,6 +699,11 @@ func (a *dexAccount) setupCryptoV2(creds *db.PrimaryCredentials, crypter encrypt
 
 func (a *dexAccount) resetKeys() {
 	a.keyMtx.Lock()
+	if len(a.encKey) > 0 {
+		// TODO: Should only increment if the current keys were derived
+		// from a seed.
+		a.hdKeyIndex++ // increment this counter to ensure next derived hd key differs from this
```

Comment:
One gotcha that's not so obvious is that `genChild` in `setupCryptoV2` may end up using a higher key index than the one provided in its input argument since it increments if `Child` errors.  This is supposedly extremely rare but possible.  The result is that when we loop through keys in both `DiscoverAccount` and `registerHD` and increment the key, we may be incrementing to the one that was actually just tried by `genChild` if it has incremented it internally.  I do not thing this is a problem however as it will just check the same key twice and then go on to the next, but it's worth noting.